theano/gof/compilelock.py

```diff
@@ -101,7 +101,14 @@ def get_lock(lock_dir=None, **kw):
             if now - get_lock.start_time > config.compile.timeout/2:
                 lockpath = os.path.join(get_lock.lock_dir, 'lock')
                 _logger.info('Refreshing lock %s', str(lockpath))
-                refresh_lock(lockpath)
+                try:
+                    refresh_lock(lockpath)
+                except Exception:
+                    while get_lock.lock_is_enabled:
+                        release_lock()
```

Comment:
This do not hide it. The first time it appear still cause an error (I re-raise the error, I just clean up the lock state in the current process). It prevent following test of failing. So instead of causing > 1000 test failure, it would be 1 test failure. So I think my PR is good.

But I agree, we also need to fix the original. I think we need both fix. @abergeron I let you track the original problem.

I moved this logic to the refresh_lock function in a new commit.